metadata-ingestion/src/datahub/configuration/common.py

```diff
@@ -15,6 +15,58 @@
 _ConfigSelf = TypeVar("_ConfigSelf", bound="ConfigModel")
 
 
+REDACT_KEYS = {
+    "password",
+    "token",
+    "secret",
+    "options",
+    "sqlalchemy_uri",
+}
+REDACT_SUFFIXES = {
+    "_password",
+    "_secret",
+    "_token",
+    "_key",
+    "_key_id",
+}
+
+
+def _should_redact_key(key: str) -> bool:
+    return key in REDACT_KEYS or any(key.endswith(suffix) for suffix in REDACT_SUFFIXES)
+
+
+def _redact_value(value: Any) -> Any:
+    if isinstance(value, str):
+        # If it's just a variable reference, it's ok to show as-is.
+        if value.startswith("$"):
+            return value
+        return "********"
+    elif value is None:
+        return None
+    elif isinstance(value, bool):
+        # We don't have any sensitive boolean fields.
+        return value
+    elif isinstance(value, list) and not value:
+        # Empty states are fine.
+        return []
+    elif isinstance(value, dict) and not value:
+        return {}
+    else:
```

Comment:
Was it a conscious decision not to check number types?
I don't think it is an issue here, I was just wondering.

Reply:
I've only not redacted bools, env refs, and empty lists/dicts. I figured that a number could be an account ID or something and hence should be redacted in general.